docs/hazmat/primitives/aead.rst

```diff
@@ -228,6 +228,89 @@ also support providing integrity for associated data which is not encrypted.
             when the ciphertext has been changed, but will also occur when the
             key, nonce, or associated data are wrong.
 
+.. class:: AESSIV(key)
+
+    .. versionadded:: 37.0
+
+    The SIV (synthetic initialization vector) construction is defined in
+    :rfc:`5297`. Depending on how it is used, SIV allows either
+    deterministic authenticated encryption or nonce-based,
+    misuse-resistant authenticated encryption.
+
+    :param key: A 256, 384, or 512-bit key (double sized from typical AES).
+        This **must** be kept secret.
+    :type key: :term:`bytes-like`
+
+    :raises cryptography.exceptions.UnsupportedAlgorithm: If the version of
+        OpenSSL does not support AES-SIV.
+
+    .. doctest::
+
+        >>> import os
+        >>> from cryptography.hazmat.primitives.ciphers.aead import AESSIV
+        >>> data = b"a secret message"
+        >>> nonce = os.urandom(16)
+        >>> aad = b["authenticated but unencrypted data", nonce]
+        >>> key = AESSIV.generate_key(bit_length=512)  # AES256 requires 512-bit keys for SIV
+        >>> aessiv = AESSIV(key)
+        >>> ct = aessiv.encrypt(data, aad)
+        >>> aessiv.decrypt(ct, aad)
+        b'a secret message'
+
+    .. classmethod:: generate_key(bit_length)
+
+        Securely generates a random AES-SIV key.
+
+        :param bit_length: The bit length of the key to generate. Must be
+            256, 384, or 512. AES-SIV splits the key into an encryption and
+            MAC key, so these lengths correspond to AES 128, 192, and 256.
+
+        :returns bytes: The generated key.
+
+    .. method:: encrypt(data, associated_data)
+
+        .. note::
+
+            SIV performs nonce-based authenticated encryption when a component of
+            the associated data is a nonce. The final associated data in the
+            list is used for the nonce.
+
+            Random nonces should have at least 128-bits of entropy. If a nonce is
+            reused with SIV authenticity is retained and confidentiality is only
+            compromised to the extent that an attacker can determine that the
+            same plaintext (and same associated data) was protected with the same
+            nonce and key.
```

Comment:
Maybe mention that if you don't provide a nonce encryption is entirely deterministic